pkg/api/convertertov20180930preview.go

```diff
@@ -8,44 +8,44 @@ import (
 // v20180930preview.OpenShiftManagedCluster.
 func ConvertToV20180930preview(cs *OpenShiftManagedCluster) *v20180930preview.OpenShiftManagedCluster {
 	oc := &v20180930preview.OpenShiftManagedCluster{
-		ID:       &cs.ID,
-		Location: &cs.Location,
-		Name:     &cs.Name,
-		Type:     &cs.Type,
+		ID:       nilIfStringEmpty(&cs.ID),
+		Location: nilIfStringEmpty(&cs.Location),
+		Name:     nilIfStringEmpty(&cs.Name),
+		Type:     nilIfStringEmpty(&cs.Type),
 	}
 	oc.Tags = make(map[string]*string, len(cs.Tags))
 	for k := range cs.Tags {
 		v := cs.Tags[k]
-		oc.Tags[k] = &v
+		oc.Tags[k] = nilIfStringEmpty(&v)
 	}
 
 	oc.Plan = &v20180930preview.ResourcePurchasePlan{
-		Name:          &cs.Plan.Name,
-		Product:       &cs.Plan.Product,
-		PromotionCode: &cs.Plan.PromotionCode,
-		Publisher:     &cs.Plan.Publisher,
+		Name:          nilIfStringEmpty(&cs.Plan.Name),
+		Product:       nilIfStringEmpty(&cs.Plan.Product),
+		PromotionCode: nilIfStringEmpty(&cs.Plan.PromotionCode),
+		Publisher:     nilIfStringEmpty(&cs.Plan.Publisher),
 	}
 
 	provisioningState := v20180930preview.ProvisioningState(cs.Properties.ProvisioningState)
 	oc.Properties = &v20180930preview.Properties{
 		ProvisioningState: &provisioningState,
```

Comment:
Hrm, I think provisioningState, count, vmSize, osType, and role also need to be handled the same way?

Reply:
Amended the commit with provisioningState, vmSize, osType and role *""→nil conversion. Also changed the test case a bit to include internal fields of AgentPoolProfile.

Reply:
nvm, forgot about converttoadmin.go and _test. BRB with more amends.

Reply:
amended with converttoadmin.go changes